grails-app/controllers/org/pih/warehouse/product/ProductController.groovy

```diff
@@ -1103,6 +1104,42 @@ class ProductController {
         }
         render(view: "addDocument", model: [productInstance: productInstance, documentInstance: documentInstance])
     }
+
+    def showMergeProductDialog = {
+        Product primaryProduct = Product.get(params.primaryProduct)
+        render(template: params.template, model: [ primaryProduct: primaryProduct ])
+    }
+
+    def mergeProduct = {
+        Boolean enabled = ConfigurationHolder.config.openboxes.mergeProducts.enabled
+        if (!enabled) {
+            flash.message = "${warehouse.message(code: 'product.mergeProducts.notEnabled.label')}"
+            redirect(action: "list")
+            return
+        }
+         Product primaryProduct = Product.get(params.primaryProduct)
+         Product obsoleteProduct = Product.get(params.obsoleteProduct)
+
+        String errors = ""
+
+        if (!primaryProduct) {
+            errors = "${warehouse.message(code: 'default.not.found.message', args: [warehouse.message(code: 'product.label', default: 'Product'), primaryProduct.id])}"
```

Comment:
You could add to the string instead of assign (plus add `\n` at the end of this message) and you could potentially have both missing products in one message if that would be the case.